packages/notification-center/src/hooks/use-unseen-count.hook.ts

```diff
@@ -1,8 +1,51 @@
-import { useContext } from 'react';
-import { UnseenCountContext } from '../store/unseen-count.context';
+import { useEffect } from 'react';
+import { useQuery, useQueryClient, UseQueryOptions } from '@tanstack/react-query';
 
-export function useUnseenCount() {
-  const { unseenCount, setUnseenCount } = useContext(UnseenCountContext);
+import type { ICountData } from '../shared/interfaces';
+import { FEED_UNSEEN_COUNT_QUERY_KEY, INFINITE_NOTIFICATIONS_QUERY_KEY, UNSEEN_COUNT_QUERY_KEY } from './queryKeys';
+import { useNovuContext } from './use-novu-context.hook';
 
-  return { unseenCount, setUnseenCount };
-}
+const dispatchUnseenCountEvent = (count: number) => {
+  document.dispatchEvent(new CustomEvent('novu:unseen_count_changed', { detail: count }));
+};
+
+export const useUnseenCount = ({ onSuccess, ...restOptions }: UseQueryOptions<ICountData, Error, ICountData> = {}) => {
+  const { apiService, socket, isSessionInitialized } = useNovuContext();
+  const queryClient = useQueryClient();
+
+  useEffect(() => {
+    if (!socket) {
+      return () => {};
+    }
+
+    socket.on('unseen_count_changed', (data?: { unseenCount: number }) => {
+      if (Number.isInteger(data?.unseenCount)) {
+        queryClient.setQueryData<{ count: number }>(UNSEEN_COUNT_QUERY_KEY, (oldData) => ({
+          count: data?.unseenCount ?? oldData.count,
+        }));
+        queryClient.refetchQueries(INFINITE_NOTIFICATIONS_QUERY_KEY, {
+          exact: false,
+        });
+        queryClient.refetchQueries(FEED_UNSEEN_COUNT_QUERY_KEY, {
+          exact: false,
+        });
+        dispatchUnseenCountEvent(data.unseenCount);
+      }
+    });
+
+    return () => {
+      socket.off('unseen_count_changed');
+    };
+  }, [socket, queryClient]);
+
+  const result = useQuery<ICountData, Error, ICountData>(UNSEEN_COUNT_QUERY_KEY, () => apiService.getUnseenCount(), {
+    ...restOptions,
+    enabled: isSessionInitialized,
+    onSuccess: (data) => {
+      dispatchUnseenCountEvent(data.count);
+      onSuccess?.(data);
+    },
+  });
+
+  return result;
+};
```

Comment:
Fetch the `unseen` count from the API first. 
When the socket is initialized then listen to the `unseen_count_changed` event and:
- update the query data for the initial unseen count
- refetch all notifications
- refetch feed unseen count